Assets/MRTK/Providers/OpenXR/Scripts/MicrosoftArticulatedHand.cs

```diff
@@ -76,20 +86,80 @@ public override void UpdateController(InputDevice inputDevice)
                 if (inputDevice.TryGetFeatureValue(CommonUsages.devicePosition, out Vector3 _))
                 {
                     base.UpdateController(inputDevice);
+
+                    // We've gotten device data from the platform, don't attempt to infer other input actions
+                    // from the hand joint data
+                    receivingDeviceInputs = true;
                 }
                 else
                 {
                     UpdateHandData(inputDevice);
 
+                    // Updating the Index finger pose right after getting the hand data
+                    // regardless of whether device data is present
                     for (int i = 0; i < Interactions?.Length; i++)
                     {
-                        switch (Interactions[i].AxisType)
+                        var interactionMapping = Interactions[i];
+                        switch (interactionMapping.InputType)
                         {
-                            case AxisType.SixDof:
-                                UpdatePoseData(Interactions[i], inputDevice);
+                            case DeviceInputType.IndexFinger:
+                                handDefinition?.UpdateCurrentIndexPose(Interactions[i]);
```

Comment:
nit, since it's cached:

```suggestion
                                handDefinition?.UpdateCurrentIndexPose(interactionMapping);
```